common/src/abi.rs

```diff
@@ -175,6 +176,27 @@ pub fn format_token(param: &Token) -> String {
     }
 }
 
+/// Formats an integer as a string together with its exponential notation.
+pub fn format_uint_with_exponential_hint(num: &U256) -> String {
+    if num.lt(&U256::from(10_000)) {
+        return num.to_string()
+    }
+
+    let stringified = num.to_string();
```

Comment:
the display impl truncates I believe

Reply:
Not sure what you mean, sorry, could you expand?